zulipterminal/ui_tools/views.py

```diff
@@ -1079,6 +1081,19 @@ def __init__(self, controller: Any, stream_id: int) -> None:
         self.controller = controller
         stream = controller.model.stream_dict[stream_id]
 
+        def timestamp_to_local_datetime(timestamp: int) -> str:
+            local_timezone = tzlocal.get_localzone()
+            local_time = datetime.fromtimestamp(timestamp, local_timezone)
+
+            formatted_datetime = local_time.strftime("%B %d, %Y")
+            return formatted_datetime
```

Comment:
Hmm. Seems like the class parametrization was not a very good idea afterall. But, I am not sure how else to add tests to this commit without making the subsequent commits dependent on it.

If you want to centralize this datetime conversion first, I can put it up on a separate PR inside maybe `helper.py` and then refactor this commit to use that function. If you have any other idea in mind, it would help me to know. :)

Reply:
The centralized preferred formatting is merged now as a Model method.